### PR TITLE
Add array destructuring documentation to array syntax section

### PR DESCRIPTION
**Pull request by rask**
This redoes the original PR I made against this repo, but which got stale: #19 

Adds basic syntax documentation for array destructuring.

### CODE REVIEW

File: language/types/array.xml [hunk 1]
```diff
@@ -488,11 +488,46 @@ Array
...
    </note>
 
   </sect3>
+
+  <sect3 xml:id="language.types.array.syntax.destructuring">
+   <title>Array destructuring</title>
+
+   <note>
+    <para>
+     Array destructuring is available only in PHP versions 7.1 and newer.
```
Comment (Girgias) [inline language/types/array.xml:502]:
```suggestion
     Array destructuring is available as of PHP 7.1.0.
```

File: language/types/array.xml [hunk 2]
```diff
@@ -488,11 +488,46 @@ Array
...
+    </para>
+   </note>
+
+   <para>
+    Arrays can be destructured symmetrically. What this means is you can
+    define local scope variables using values from an existing array.
+   </para>
+
+   <para>
+    This feature is similar to what the <function>list</function> construct offers.
```
Comment (Girgias) [inline language/types/array.xml:512]:
```suggestion
    This feature is equivalent to the <function>list</function> construct.
```

**Review by Girgias — CHANGES_REQUESTED** [review]
I'm concerned that the array type pages is maybe not the best place as it's already a massive and dense page, but this has nothing to do with the PR really.

**Comment by afilina** [comment]
Pinging @rask to address suggestions. Looks like this PR fell through the cracks.

**Review by cmb69 — CHANGES_REQUESTED** [review]
IMO, we should not duplicate the description of `list()` (array destructoring is just syntax sugar for that, like literal arrays are syntax sugar for `array()`.  Instead we should clarify on the list man page, that the shorthand syntax is supported as of PHP 7.1.0 and maybe give some examples. It may make sense to add a short note to the array docs, telling users that literal arrays used as lvalues are actually destructuring and link to list (just in case somebody is mislead by that construct).

**Review by gallogit — CHANGES_REQUESTED** [review]
Corregir errores del sistema para licencias en México multriples

**Review by gallogit — APPROVED** [review]
Corregir error

**Comment by afilina** [comment]
The comments were not addressed by the author in over a year, despite a reminder in January. Should we consider this PR abandoned and close it? @rask

**Comment by kamil-tekiela** [comment]
I have opened a new PR recently which tried to address the open points in this PR and add even more information to the page. I believe my PR supersedes this one.

**Comment by cmb69** [comment]
Oh, right, there is PR #796, so I'm closing this one.

**Pull request closed without merging**